sql/analyzer/indexes.go

```diff
@@ -349,6 +348,35 @@ func getIndexes(
 			lookup:  lookup,
 			expr:    e,
 		}
+	case *expression.MatchAgainst:
+		// AFAICT, it looks like Full-Text index searches actually do a full table scan, and will return different
+		// results depending on the values contained in all rows. This means that we don't need to handle the filter
+		// ourselves, but we still want to take advantage of the ease of grabbing the correct index here, so that's all
+		// we do.
```

Comment:
Full-text index searches can't be full table scans of the matched columns, right? I think the whole point of the index is to not do that. If you have a table with a single text column, it has 8 rows, and 7 of those rows have 8GB of data in their text column each, and the 8th row has the string `"This is a unique string across the corpus: zkjdfurhcnbds"`, surely a `MATCH (text_col) AGAINST ("zkjdfurhcnbds")` doesn't scan the 56GB of extraneous TEXT data in the table?

Or am I misinterpreting this comment here?